aptos-move/move-examples/token_objects/sources/collection.move

```diff
@@ -0,0 +1,439 @@
+    use std::error;
+    use std::option::{Self, Option};
+    use std::signer;
+    use std::string::{Self, String};
+
+    use aptos_framework::object::{Self, CreatorRef, ObjectId};
+
+    friend token_objects::token;
+
+    /// The collections supply is at its maximum amount
+    const EEXCEEDS_MAX_SUPPLY: u64 = 0;
+    /// The collection does not exist
+    const ECOLLECTION_DOES_NOT_EXIST: u64 = 1;
+    /// The provided signer is not the creator
+    const ENOT_CREATOR: u64 = 2;
+    /// Attempted to mutate an immutable field
+    const EFIELD_NOT_MUTABLE: u64 = 3;
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    /// Represents the common fields for a collection.
+    struct Collection has key {
+        /// The creator of this collection.
+        creator: address,
+        /// A brief description of the collection.
+        description: String,
+        /// Determines which fields are mutable.
+        mutability_config: MutabilityConfig,
+        /// An optional categorization of similar token.
+        name: String,
+        /// The Uniform Resource Identifier (uri) pointing to the JSON file stored in off-chain
+        /// storage; the URL length will likely need a maximum any suggestions?
+        uri: String,
+    }
+
+    /// This config specifies which fields in the TokenData are mutable
+    struct MutabilityConfig has copy, drop, store {
+        description: bool,
+        uri: bool,
+    }
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    /// The royalty of a token within this collection -- this optional
+    struct Royalty has drop, key {
+        numerator: u64,
+        denominator: u64,
+        /// The recipient of royalty payments. See the `shared_account` for how to handle multiple
+        /// creators.
+        payee_address: address,
+    }
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    /// Aggregable supply tracker, this is can be used for maximum parallel minting but only for
+    /// for uncapped mints. Currently disabled until this library is in the framework.
+    struct AggregableSupply has key {
+    }
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    /// Fixed supply tracker, this is useful for ensuring that a limited number of tokens are minted.
+    struct FixedSupply has key {
+        current_supply: u64,
+        max_supply: u64,
+    }
+
+    public fun create_fixed_collection(
+        creator: &signer,
+        description: String,
+        max_supply: u64,
+        mutability_config: MutabilityConfig,
+        name: String,
+        royalty: Option<Royalty>,
+        uri: String,
+    ): CreatorRef {
+        let collection_seed = derive_collection_id_seed(&name);
+        let creator_ref = object::create_named_object(creator, collection_seed);
+        let object_signer = object::generate_signer(&creator_ref);
+
+        let collection = Collection {
```

Comment:
This could be captured as follows (after the latest PRs land):

```
fun inline object_helpers::create<T>(ref: &CreatorRef, x: T) { 
  move_to(object::generate_signer(ref), x) 
}
```

Reply:
looking forward to it.